ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -371,14 +373,38 @@ long getCapacityInBytes() {
   /**
    * Fetches a list of {@link LogSegment} names whose entries don't over lap with {@link Journal}. Returns {@code null}
    * if there aren't any
-   * @return list of {@link LogSegment} names whose entries don't over lap with {@link Journal}. {@code null}
+   * @return a list of {@link LogSegment} names whose entries don't over lap with {@link Journal}. {@code null}
    * if there aren't any
    */
   List<String> getLogSegmentsNotInJournal() throws StoreException {
     checkStarted();
     return index.getLogSegmentsNotInJournal();
   }
 
+  /**
+   * Fetches segment capacity of a single log segment
+   * @return the segment capacity of a single log segment
+   */
+  long getSegmentCapacity() {
+    return log.getSegmentCapacity();
+  }
+
+  /**
+   * Fetches segment header size of a single log segment
+   * @return the segment header size of a single log segment
+   */
+  long getLogSegmentHeaderSize() {
```

Comment:
Makes sense. I am looking at something like this. Lets discuss offline and take a call.

```
interface CostBenefitCalculator {

  CompactionDetails getCompactionDetails(long totalCapacity, long usedCapacity, long segmentCapacity,
      long segmentHeaderSize, List<String> logSegmentsNotInJournal, BlobStoreStats blobStoreStats)
      throws StoreException;

  // goes over all candidates who doesn't over lap with jounral and finds the best candidate
  CostBenefitInfo getBestCandidateToCompact(NavigableMap<String, Long> validDataPerLogSegments, long segmentCapacity,
      long segmentHeaderSize, long errorMargin);

  // finds the cost benefit ratio for a given candidate
  CostBenefitInfo getCostBenefitInfo(String firstLogSegmentName, String lastLogSegmentName,
      NavigableMap<String, Long> validDataSizePerLogSegment, long segmentCapacity, long segmentHeaderSize,
      long errorMargin);
}

class SimpleCostBenefitCalculator implements CostBenefitCalculator{

 // all method definitions
 .
 .
 .
 }

In CompactionManager: 
SimpleCostBenefitCalculator calculator = new SimpleCostBenefitCalculator()
CompactionDetails details = store.computeCostBenefitAndGetCompactionDetails(calculator);

In BlobStore:
CompactionDetails computeCostBenefitAndGetCompactionDetails(CostBenefitCalculator calculator){
    return calculator.getCompactionDetails(totalCapacity(),usedCapacity, segmentCapacity,
      segmentHeaderSize, logSegmentsNotInJournal, blobStoreStats);
}
```